leshan-core/src/main/java/org/eclipse/leshan/core/request/execute/Arguments.java

```
@@ -0,0 +1,260 @@
+     * CHAR = "!" / %x23-26 / %x28-5B / %x5D-7E
+     * }
+     * </pre>
+     * 
+     * @param arglist String contains arguments to parse.
+     * @return {@link Arguments} represents list of arguments.
+     * @throws InvalidArgumentException if text has invalid format.
+     */
+    static Arguments parse(String arglist) throws InvalidArgumentException {
+        List<Argument> argumentList = new ArrayList<>();
+
+        if (arglist != null && !arglist.isEmpty()) {
+            int beginProbe = 0;
+            int validBegin = 0;
+            while (true) {
+                int separatorIndex = arglist.indexOf(',', beginProbe);
+                if (separatorIndex == -1) {
+                    break;
+                }
+                String argument = arglist.substring(validBegin, separatorIndex);
+
+                ArgumentParser argumentParser = new ArgumentParser(argument, arglist);
+                if (argumentParser.isValid()) {
+                    argumentList.add(argumentParser.parse());
+                    validBegin = separatorIndex + 1;
+                }
+                beginProbe = separatorIndex + 1;
+            }
+
+            String argument = arglist.substring(beginProbe);
+            ArgumentParser argumentParser = new ArgumentParser(argument, arglist);
+            argumentList.add(argumentParser.parse());
+        }
+
+        return new Arguments(argumentList);
+    }
+
+    private static class ArgumentParser {
+        private final String digitPart;
+        private final String valueDecorated;
+        private final String content;
+
+        public ArgumentParser(String argument, String content) {
+            this.content = content;
+            String[] keyValue = argument.split("=", 2);
+            digitPart = keyValue[0];
+            valueDecorated = keyValue.length == 1 ? null : keyValue[1];
+        }
+
+        private void validate() throws InvalidArgumentException {
```

I'm not sure we need a `validate` + `isValid` maybe we just raise InvalidArgumentException in `parse`

---

Ok, I can inline validation in parse.